pkg/controller/controllercmd/builder.go

```diff
@@ -157,6 +159,14 @@ func (b *ControllerBuilder) WithLeaderElection(leaderElection configv1.LeaderEle
 	return b
 }
 
+// WithLeaderElectionResourceLock sets the resource lock based on the options in the
+// k8s.io/client-go/tools/leaderelection/resourcelock package. A safe default is configmapsleases.
+func (b *ControllerBuilder) WithLeaderElectionResourceLock(resourceLock string) *ControllerBuilder {
+	b.leaderElectionResourceLock = resourceLock
+
+	return b
+}
```

Comment:
As mentioned in my other comment, we shouldn't allow every type of lock since some of them are not supported anymore.